File: src/main/java/com/enzo/testemuralis/services/viacep/dtos/EnderecoViacep.java
package com.enzo.testemuralis.services.viacep.dtos;

public record EnderecoViacep(
        String cep,
        String logradouro,
        String complemento,
        String bairro,
        String localidade,
        String uf,
        String ibge,
        String gia,
        String ddd,
        String siafi) {
}

File: dotnet/TesteMuralis/Services/Viacep/Dto/EnderecoViacep.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TesteMuralis.Services.Viacep.Dto
{
    public class EnderecoViacep
    {
        public String cep { get; set; }
        public String logradouro { get; set; }
        public String complemento { get; set; }
        public String bairro { get; set; }
        public String localidade { get; set; }
        public String uf { get; set; }
        public String ibge { get; set; }
        public String gia { get; set; }
        public String ddd { get; set; }
        public String siafi { get; set; }
    }
}